osu.Game.Rulesets.Mania/UI/ManiaPlayfield.cs

```diff
@@ -157,82 +99,45 @@ public ManiaPlayfield(int columnCount)
         private void invertedChanged(bool newValue)
         {
             Scale = new Vector2(1, newValue ? -1 : 1);
-            judgements.Scale = Scale;
-        }
-
-        [BackgroundDependencyLoader]
-        private void load(OsuColour colours)
-        {
-            normalColumnColours = new List<Color4>
-            {
-                colours.RedDark,
-                colours.GreenDark
-            };
-
-            specialColumnColour = colours.BlueDark;
-
-            // Set the special column + colour + key
-            foreach (var column in Columns)
-            {
-                if (!column.IsSpecial)
-                    continue;
 
-                column.AccentColour = specialColumnColour;
-            }
-
-            var nonSpecialColumns = Columns.Where(c => !c.IsSpecial).ToList();
-
-            // We'll set the colours of the non-special columns in a separate loop, because the non-special
-            // column colours are mirrored across their centre and special styles mess with this
-            for (int i = 0; i < Math.Ceiling(nonSpecialColumns.Count / 2f); i++)
+            foreach (var single in listColumnStages)
             {
-                Color4 colour = normalColumnColours[i % normalColumnColours.Count];
-                nonSpecialColumns[i].AccentColour = colour;
-                nonSpecialColumns[nonSpecialColumns.Count - 1 - i].AccentColour = colour;
+                single.Judgements.Scale = Scale;
             }
         }
 
         public override void OnJudgement(DrawableHitObject judgedObject, Judgement judgement)
         {
             var maniaObject = (ManiaHitObject)judgedObject.HitObject;
-            columns[maniaObject.Column].OnJudgement(judgedObject, judgement);
+            int column = maniaObject.Column;
+            Columns[column].OnJudgement(judgedObject, judgement);
 
-            judgements.Clear();
-            judgements.Add(new DrawableManiaJudgement(judgement)
-            {
-                Anchor = Anchor.Centre,
-                Origin = Anchor.Centre,
-            });
+            getFallDownControlContainerByActualColumn(column).AddJudgement(judgement);
         }
 
-        /// <summary>
-        /// Whether the column index is a special column for this playfield.
-        /// </summary>
-        /// <param name="column">The 0-based column index.</param>
-        /// <returns>Whether the column is a special column.</returns>
-        private bool isSpecialColumn(int column)
+        public override void Add(DrawableHitObject h) => Columns.ElementAt(((ManiaHitObject)h.HitObject).Column).Add(h);

```

Comment:
This (along with OnJudgement above) should be done in the following three steps:

1. Make this call `Add` on the stage which the column belongs to, with the hitobject. You can use the `getStageByColumn` method here.
2. Have the stage's OnJudgement do the `AddJudgement` stuff.
3. Remove the OnJudgement call for the column, that should be handled by the column itself.